rules/codelingo/go/reassigned-error/codelingo.yaml

```diff
@@ -14,8 +14,45 @@ funcs:
 tenets:
 - name: reassigned-error
   actions:
+      codelingo/docs:
+        title: Check `err` Before Variable Re-Assignment
+        body: | 
+          When re-assigning a variable in Go, it is convention to check the `err` prior to
+          doing so. First, check that the `err` is not equal to `nil` and then re-assign
+          the variable to avoid a panic.
```

Comment:
This is sheer pedantry, but:

* a non-nil error can be perfectly legitimate (e.g. deleting something that didn't exist)
* a panic is probably the best possible outcome in this scenario (because at least you _stop executing_ instead of continuing on with bad data, e.g. if you forget to check the error from your authorization component and hand over people's private data to anyone who asks for it)

Reply:
(I'd probably compress it to something like "Always check errors before executing dependent code.")